File: library/src/main/java/com/grosner/dbflow/sql/migration/BaseMigration.java
package com.grosner.dbflow.sql.migration;

import android.database.sqlite.SQLiteDatabase;

/**
 * Author: andrewgrosner
 * Contributors: { }
 * Description: Provides the base implementation of {@link com.grosner.dbflow.sql.migration.Migration} with
 * only {@link #migrate(android.database.sqlite.SQLiteDatabase)} needing to be implemented.
 */
public abstract class BaseMigration implements Migration {


    @Override
    public void onPreMigrate() {

    }

    @Override
    public abstract void migrate(SQLiteDatabase database);

    @Override
    public void onPostMigrate() {

    }
}

File: dbflow/src/main/java/com/raizlabs/android/dbflow/migration/BaseMigration.kt
package com.raizlabs.android.dbflow.migration

import com.raizlabs.android.dbflow.database.DatabaseWrapper

/**
 * Description: Provides the base implementation of [com.raizlabs.android.dbflow.sql.migration.Migration] with
 * only [Migration.migrate] needing to be implemented.
 */
abstract class BaseMigration : Migration {


    override fun onPreMigrate() {

    }

    abstract override fun migrate(database: DatabaseWrapper)

    override fun onPostMigrate() {

    }
}
